cmd/kpod/stop.go

```diff
@@ -54,26 +62,41 @@ func stopCmd(c *cli.Context) error {
 
 	logrus.Debugf("Stopping containers with timeout %d", stopTimeout)
 
+	var filterFuncs []libpod.ContainerFilter
+	var containers []*libpod.Container
 	var lastError error
-	for _, container := range c.Args() {
-		ctr, err := runtime.LookupContainer(container)
+
+	if c.Bool("all"){
+		// only get running containers
+		filterFuncs = append(filterFuncs, func(c *libpod.Container) bool {
+			state, _ := c.State()
+			return state == libpod.ContainerStateRunning
+		})
+		containers, err = runtime.GetContainers(filterFuncs...)
 		if err != nil {
-			if lastError != nil {
-				fmt.Fprintln(os.Stderr, lastError)
+			return errors.Wrapf(err, "unable to get running containers")
+		}
+	} else {
+		for _, i := range args {
+			container, err := runtime.LookupContainer(i)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
```

Comment:
This is wrong it should print lasterror if it is set and then reset it in the next line.    You code will end up printing the lasterror twice.

Reply:
```
 			if lastError != nil {
 				fmt.Fprintln(os.Stderr, lastError)
 			}
```

Reply:
fixed, thank you